LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/Processors/CupsProtocolMessageProcessor.cs

```diff
@@ -33,14 +36,41 @@ internal class CupsProtocolMessageProcessor : ICupsProtocolMessageProcessor
 
         public async Task HandleUpdateInfoAsync(HttpContext httpContext, CancellationToken token)
         {
+            // checking content length
+            var contentLength = httpContext.Request.ContentLength;
+            if (contentLength is null)
+            {
+                LogAndSetBadRequest(httpContext, null, "Request is not specifying a Content-Length.");
+                return;
+            }
+            if (contentLength > MaximumAllowedContentLength)
+            {
+                LogAndSetBadRequest(httpContext, null, "Request body is exceeding the maximum content-length limit of {MaximumAllowedContentLength}.", MaximumAllowedContentLength);
+                return;
+            }
+
+            // reading the input stream
             using var reader = new StreamReader(httpContext.Request.Body);
-            var input = await reader.ReadToEndAsync();
+            var inputChars = new char[(int)contentLength];
```

Comment:
This won't work. The `Content-Length` header in HTTP is the size in bytes, not characters! It might work incidentally with 8-bit character sets.

You should read the bytes to a buffer then get the character set encoding from the headers and use that to convert bytes to string.

Reply:
Fixed as discussed